docs/assessment.md

```diff
@@ -226,42 +226,60 @@ Students are able to see their groupmates' UIDs, which can become a point of con
+Each question blocks all later questions until its `advanceScorePerc` is met. In the above example, `q1` blocks all later questions until the student has scored at least 50% on it. Then `q2` blocks all later questions until the student has a perfect score on it. Once a student gets past `q2`, both `q3` and `q4` are immediately available because `q3` does not do any blocking. Finally, `q4` blocks the remaining `q5` until the student has an 80% score on it. The `advanceScorePerc` attribute on `q5` is irrelevant because there are no questions after it.
 
-If a student uses all of their attempts on a question and cannot submit any more attempts, the next question will automatically unlock, no matter what score they earned on the previous question. This is to prevent students from getting permanently stuck on an assessment, unable to receive further credit.
+The relevant score for comparing to `advanceScorePerc` is the student's _highest submission score_ for the question, not their percentage score on the question overall. For example, suppose `q1` above has `"points": [10, 4, 2, 1]`. Then a student who makes a 50%-correct submission on their second attempt will unblock the question, even though they only score 2 points out of 10 on the question (50% of the 4-point second-chance value).
 
-### Note about exam-type assessments and in-order questions
+An `advanceScorePerc` can also be set on the `zone` or `assessment` level, which will act as a default for all questions in that zone or assessment. For example, the following configuration is equivalent to the above:
 
-The `advanceScorePerc` attribute is intended to be used in [group work](#enabling-group-work-for-collaborative-assessments) and assessment types which are indirectly supported, such as worksheets (see [multiple instance assessments](#multiple-instance-versus-single-instance-assessments)). In the interest of allowing students to best demonstrate their knowledge of course material, we **strongly** discourage the use of this feature in actual exams.
+```json
+{
+  "zones": [
+    {
+      "advanceScorePerc": 100,
+      "questions": [
+        { "id": "q1", "advanceScorePerc": 50 },
+        { "id": "q2" },
+        { "id": "q3", "advanceScorePerc": 0 },
+        { "id": "q4", "advanceScorePerc": 80 },
+        { "id": "q5" }
+      ]
+    }
+  ]
+}
+```
+
+In the example above, `q2` and `q5` will have an `advanceScorePerc` of 100 because the zone-level attribute is used as a default.
+
+Note that an `advanceScorePerc` of 0 is equivalent to not having the attribute at all.
+
+If a student uses all of their attempts on a question and cannot submit any more attempts, that question will automatically unblock, no matter what score they earned on it. This is to prevent students from getting permanently stuck on an assessment, unable to receive further credit.
+
+### Warning about in-order questions and high-stakes exams
```

Comment:
What about exam's that randomize the order of questions?

Reply:
I added the following. Is this what you meant? https://github.com/PrairieLearn/PrairieLearn/blob/e207091508b9e347aea9aefd839db2a16a6a42d3/docs/assessment.md?plain=1#L278

Reply:
Yes, that clarifies what is implemented. That said, it is probably a good idea to warn instructors that if there is a specific order required for this setting (i.e., if they use this option to ensure they work in a particular order) they should follow https://prairielearn.readthedocs.io/en/latest/assessment/#changing-question-order-randomization.

Reply:
Sorry, read wrong. It will be auto score only, and null if there are no auto points.

Reply:
Added: https://github.com/PrairieLearn/PrairieLearn/blob/9c28fc98b808a98cc84da2079f71c4db7e0de9d4/docs/assessment.md?plain=1#L278